pallets/did/src/lib.rs

```diff
@@ -165,27 +165,55 @@ pub mod pallet {
 			+ Dispatchable<Origin = <Self as Config>::Origin, PostInfo = PostDispatchInfo>
 			+ GetDispatchInfo
 			+ DeriveDidCallAuthorizationVerificationKeyRelationship;
+
 		/// Type for a DID subject identifier.
 		type DidIdentifier: Parameter + Default + DidVerifiableIdentifier;
+
 		#[cfg(not(feature = "runtime-benchmarks"))]
 		/// Origin type expected by the proxied dispatchable calls.
 		type Origin: From<DidRawOrigin<DidIdentifierOf<Self>>>;
+
 		#[cfg(feature = "runtime-benchmarks")]
 		type Origin: From<RawOrigin<DidIdentifierOf<Self>>>;
 		type EnsureOrigin: EnsureOrigin<Success = DidIdentifierOf<Self>, <Self as frame_system::Config>::Origin>;
 		/// Overarching event type.
 		type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
+
+		/// Maximum number of total public keys which can be stored per DID key
+		/// identifier. This includes the ones currently used for
+		/// authentication, key agreement, attestation, and delegation. It also
+		/// contains old attestation keys which cannot issue new attestations
+		/// but can still be used to verify previously issued attestations.
+		#[pallet::constant]
+		type MaxPublicKeysPerDidKeyIdentifier: Get<u32>;
+
 		/// Maximum number of key agreement keys that can be added in a creation
 		/// or update operation.
 		#[pallet::constant]
 		type MaxNewKeyAgreementKeys: Get<u32>;
+
+		/// Maximum number of total key agreement keys that can be stored for a
+		/// DID subject.
+		///
+		/// Should be greater than `MaxNewKeyAgreementKeys`.
+		#[pallet::constant]
+		type MaxTotalKeyAgreementKeys: Get<u32> + Debug + Clone + PartialEq;
+
 		/// Maximum number of keys that can be removed in an update operation.
 		#[pallet::constant]
 		type MaxVerificationKeysToRevoke: Get<u32>;
+
+		/// Maximum number of old attestation keys which can still be used to
+		/// verify previously issued attestations but cannot create new
+		/// attestations.
+		#[pallet::constant]
+		type MaxOldAttestationKeys: Get<u32> + Debug + Clone + PartialEq;
```

Comment:
I think this is not necessary. We have `MaxPublicKeysPerDidKeyIdentifier `/`MaxPublicKeysPerDid` to bound the max number of public keys that can be stored, and we have `MaxVerificationKeysToRevoke` to bound the maximum number of keys that can be removed in a DID update operation. As the public keys already include the past attestation keys, I would say we do not need this parameter here.